crates/masonry/src/text2/edit.rs

```diff
@@ -52,27 +52,31 @@ impl EditableText for String {
 
 /// A region of text which can support editing operations
 pub struct TextEditor<T: EditableText> {
-    selection: TextWithSelection<T>,
+    inner: TextWithSelection<T>,
     /// The range of the preedit region in the text
     /// This is currently unused
-    _preedit_range: Option<Range<usize>>,
+    preedit_range: Option<Range<usize>>,
 }
 
 impl<T: EditableText> TextEditor<T> {
     pub fn new(text: T, text_size: f32) -> Self {
         Self {
-            selection: TextWithSelection::new(text, text_size),
-            _preedit_range: None,
+            inner: TextWithSelection::new(text, text_size),
+            preedit_range: None,
         }
     }
 
+    pub fn reset_preedit(&mut self) {
+        self.preedit_range = None;
```

Comment:
I think this should just be part of `set_text`; we might need to "redefine" it directly on the `TextEditor`

Reply:
Maybe, but I don't see how to do that cleanly right now... possibly out of intellectual laziness. :+ )